spec/object_spec.js

```diff
@@ -111,6 +115,25 @@ describe('Transis.Object', function() {
         });
       });
 
+      describe('with a function as the default option', function() {
+        it('returns the value of calling the function', function() {
+          expect(t.funcDef).toEqual([]);
+        });
+
+        it('invokes the function once per instance', function() {
+          var t2 = new Test;
+          expect(t.funcDef).toBe(t.funcDef);
+          expect(t2.funcDef).toBe(t2.funcDef);
+          expect(t2.funcDef).not.toBe(t.funcDef);
+        });
+
+        it('returns the set value', function() {
+          expect(t.funcDef).toEqual([]);
+          t.funcDef = ['foo'];
+          expect(t.funcDef).toEqual(['foo']);
+        });
```

Comment:
@burrows Maybe a bit of an edgecase, but I would have thought, as a user, a new default value is generated (default function called) everytime a default is needed... not "needed and per instance".

```js
fit('should invoke the function every time a default is needed?', function() {
  const originalDefaultValue = t.funcDef;
  expect(t.funcDef).toEqual([]);

  t.funcDef = ['A new beginning'];
  expect(t.funcDef).toEqual(['A new beginning']);

  t.funcDef = undefined;
  expect(t.funcDef).toEqual([]);

  originalDefaultValue.push('I cUT yOu');
  expect(t.funcDef).toEqual([]); // Fail?
});
```

I don't think the above is likely to come up, but wanted to know your thoughts.